src/portfft/common/workgroup.hpp

```diff
@@ -145,16 +149,42 @@ __attribute__((always_inline)) inline void dimension_dft(
     if (working) {
       if (layout_in == detail::layout::BATCH_INTERLEAVED) {
         global_data.log_message_global(__func__, "loading transposed data from local to private memory");
-        detail::strided_view local_view{
-            loc, std::array{1, stride_within_dft, max_num_batches_in_local_mem},
-            std::array{2 * wi_id_in_fft * fact_wi, 2 * (j_inner + j_outer * outer_stride), 2 * batch_num_in_local}};
-        copy_wi<2>(global_data, local_view, priv, fact_wi);
+        if (storage == complex_storage::INTERLEAVED_COMPLEX) {
+          detail::strided_view local_view{
+              loc, std::array{1, stride_within_dft, max_num_batches_in_local_mem},
+              std::array{2 * wi_id_in_fft * fact_wi, 2 * (j_inner + j_outer * outer_stride), 2 * batch_num_in_local}};
+          copy_wi<2>(global_data, local_view, priv, fact_wi);
+        } else {
+          detail::strided_view local_real_view{
+              loc, std::array{1, stride_within_dft, max_num_batches_in_local_mem},
+              std::array{wi_id_in_fft * fact_wi, j_inner + j_outer * outer_stride, batch_num_in_local}};
```

Comment:
This repeated if-else feels a bit bulky to read. 
we are not using these views outside the scope these `if-else`, I suppose  making a utility function to avoid repeated if-else for copying between pointers would be useful (note that stride/ distance config for real and imag parts is the same). For example something like - 

```
transfer_data(MemoryType1 mem1, MemoryType2 mem2, complex_storage_config, std::array<2>(stride, distance) mem1_stride_distance, std::array<2>(stride, distance) mem2_stride_distance,) {
   create views and copy
   ....
}
```

Similarly for all other place where `if-else` is present. 
We can then use this function in other implementations  too (workitem and subgroup)

Reply:
I think refactoring this could be a bit bigger task, so it should be done in a separate PR.